ResearchKit/+Medable/ORKMDBarcodeScannerStep/ORKMDBarcodeScannerStepViewController.m

```diff
@@ -0,0 +1,291 @@
+//
+//  ORKMDBarcodeScannerStepViewController.m
+//  Medable Axon
+//
+//  Copyright (c) 2018 Medable Inc. All rights reserved.
+//
+//
+
+#import "ORKMDBarcodeScannerStepViewController.h"
+
+#import "ORKMDBarcodeScannerStep.h"
+#import "ORKMDBarcodeScannerView.h"
+
+#import "ORKInstructionStepView.h"
+#import "ORKNavigationContainerView.h"
+
+#import "ORKHelpers_Internal.h"
+#import "ORKStepHeaderView_Internal.h"
+#import "ORKStepViewController_Internal.h"
+#import "ORKNavigationContainerView_Internal.h"
+
+@interface ORKMDBarcodeScannerStepViewController () <ORKMDBarcodeScannerViewDelegate>
+
+@property (nonatomic, strong) ORKInstructionStepView *instructionStepView;
+@property (nonatomic, strong) ORKMDBarcodeScannerView *scannerView;
+
+@property (nonatomic, copy) NSString *scannerOutput;
+@property (nonatomic, assign) BOOL configurationDone;
+
+@end
+
+#pragma mark -
+
+@implementation ORKMDBarcodeScannerStepViewController
+
+- (instancetype)initWithStep:(ORKStep *)step
+{
+    self = [super initWithStep:step];
+    
+    if (self)
+    {
+        NSParameterAssert([step isKindOfClass:[ORKMDBarcodeScannerStep class]]);
+    }
+    
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    [self stepDidChange];
+    
+    self.instructionStepView = [[ORKInstructionStepView alloc] initWithFrame:self.view.bounds];
+    self.instructionStepView.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+    [self.view addSubview:self.instructionStepView];
+    
+    if (ORKMDBarcodeScannerView.isSuppported)
+    {
+        CGRect scannerFrame = self.view.bounds;
+        scannerFrame.size.height *= 0.6;
```

Comment:
Magic constant is best kept outside, but I imagine it's ok to assume 60% height of the step and this has been tested thoroughly.